packages/cli/src/cmds/assert/assertionChecker.ts

```diff
@@ -0,0 +1,21 @@
+import Assertion from './assertion';
+import Strategy from './strategy/strategy';
+import HttpRequestStrategy from './strategy/httpRequestStrategy';
+import AbortError from '../error/abortError';
+import { AppMapData } from '../../appland/types';
+
+export default class AssertionChecker {
+  private strategies: Strategy[] = [
+    new HttpRequestStrategy(),
+  ];
+
+  check(appMapData: AppMapData, assertion: Assertion): Boolean {
+    for (let strategy of this.strategies) {
+      if (strategy.supports(assertion)) {
+        return strategy.check(appMapData, assertion);
```

Comment:
Simple strategy pattern here to decouple different scope for the high-level checker.

Reply:
It would be easier for me to understand the "strategy" design if there was at least one more example of a strategy.